api/options.go

```diff
@@ -0,0 +1,99 @@
+package api
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ONSdigital/dp-cantabular-filter-flex-api/model"
+	"github.com/go-chi/chi/v5"
+	"github.com/pkg/errors"
+)
+
+func (api *API) getFilterDimensionOptions(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+
+	filterID := chi.URLParam(r, "id")
+	dimensionName := chi.URLParam(r, "name")
+
+	pageLimit, offset, err := getPaginationParams(r.URL, api.cfg.DefaultMaximumLimit)
+	if err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			http.StatusBadRequest,
+			errors.Wrap(err, "Bad Request"),
+		)
+
+	}
+
+	if pageLimit == 0 {
+		// define a reasonable default
+		// in light of bad input
+		// also slice not work with 0
+		pageLimit = 20
+	}
+
```

Comment:
You didn't want to do any pre-emptive checks for the filter/dimension etc to return the other status codes? Not wanting to throw a spanner in the works after you've done all this work but I wonder if it wouldn't of been simpler to just retrieve the entire filter object from the store, perform the checks in order then return the options from the filter..

Reply:
I believe I've returned the right status codes as per the spec with one query, so It is 6 and half a dozen for me really

Reply:
I will resolve for now if that is okay

Reply:
Ok cool